src/bin/cross-util.rs

```diff
@@ -1,252 +1,61 @@
 #![deny(missing_debug_implementations, rust_2018_idioms)]
 
-use std::path::{Path, PathBuf};
-use std::process::Command;
-
 use clap::{Parser, Subcommand};
-use cross::CommandExt;
 
-// known image prefixes, with their registry
-// the docker.io registry can also be implicit
-const GHCR_IO: &str = "ghcr.io/cross-rs/";
-const RUST_EMBEDDED: &str = "rustembedded/cross:";
-const DOCKER_IO: &str = "docker.io/rustembedded/cross:";
-const IMAGE_PREFIXES: &[&str] = &[GHCR_IO, DOCKER_IO, RUST_EMBEDDED];
+mod commands;
 
 #[derive(Parser, Debug)]
 #[clap(version, about, long_about = None)]
 struct Cli {
+    /// Toolchain name/version to use (such as stable or 1.59.0).
+    #[clap(value_parser = is_toolchain)]
+    toolchain: Option<String>,
     #[clap(subcommand)]
     command: Commands,
 }
 
 #[derive(Subcommand, Debug)]
 enum Commands {
     /// List cross images in local storage.
-    ListImages {
-        /// Provide verbose diagnostic output.
-        #[clap(short, long)]
-        verbose: bool,
-        /// Container engine (such as docker or podman).
-        #[clap(long)]
-        engine: Option<String>,
-    },
+    ListImages(commands::ListImages),
     /// Remove cross images in local storage.
-    RemoveImages {
-        /// If not provided, remove all images.
-        targets: Vec<String>,
-        /// Remove images matching provided targets.
-        #[clap(short, long)]
-        verbose: bool,
-        /// Force removal of images.
-        #[clap(short, long)]
-        force: bool,
-        /// Remove local (development) images.
-        #[clap(short, long)]
-        local: bool,
-        /// Remove images. Default is a dry run.
-        #[clap(short, long)]
-        execute: bool,
-        /// Container engine (such as docker or podman).
-        #[clap(long)]
-        engine: Option<String>,
-    },
-}
-
-#[derive(Debug, PartialOrd, Ord, PartialEq, Eq)]
-struct Image {
-    repository: String,
-    tag: String,
-    // need to remove images by ID, not just tag
-    id: String,
-}
-
-impl Image {
-    fn name(&self) -> String {
-        format!("{}:{}", self.repository, self.tag)
-    }
+    RemoveImages(commands::RemoveImages),
 }
 
-fn get_container_engine(engine: Option<&str>) -> Result<PathBuf, which::Error> {
-    if let Some(ce) = engine {
-        which::which(ce)
+fn is_toolchain(toolchain: &str) -> cross::Result<String> {
```

Comment:
Allow us to use `cross-util +nightly`, or similar, rather than `cross-util --toolchain nightly`, which is consistent with the `cross` API.